packages/contract/src/contract-factory.ts

```diff
@@ -372,6 +377,107 @@ export default class ContractFactory {
     return { waitForResult, contractId, waitForTransactionId };
   }
 
+  async deployAsBlobTxForPredicate(
+    configurableConstants: { [name: string]: unknown } = {}
+  ): Promise<{
+    waitForResult: () => Promise<{
+      transactionResult: TransactionResult<TransactionType.Blob>;
+      loaderBytecode: string;
+    }>;
+    predicateRoot: string;
+    loaderBytecode: Uint8Array;
+    loaderBytecodeHexlified: string;
+  }> {
+    /** TODO: Implement me */
+    // @ts-expect-error lol
+    return Promise.resolve({
+      waitForResult: () =>
+        Promise.resolve({
+          transactionResult: {},
+          loaderBytecode: '',
+        }),
+      predicateRoot: '',
+      loaderBytecode: new Uint8Array(),
+      loaderBytecodeHexlified: '',
+    });
```

Comment:
Return offset to generate a abi copy for the loader with new offset of the data section.
```suggestion
     return Promise.resolve({
      waitForResult: () =>
        Promise.resolve({
          transactionResult: {},
          loaderBytecode: '',
          offset: 0,
        }),
      predicateRoot: '',
      loaderBytecode: new Uint8Array(),
      loaderBytecodeHexlified: '',
      offset: 0,
    });
```